data/processors/maps.py

```diff
@@ -337,3 +337,37 @@ def _load_custom_maps():
             }
 
     return maps_out
+
+
+def add_overlay_maps(data):
+    """ Add the overlay maps to all entries where they apply """
+    with open("sources/46_overlay-maps.yaml") as f:
+        overlay_maps = yaml.safe_load(f.read())
+        
+    parent_lookup = {m["props"]["parent"]: m for m in overlay_maps}
+    parent_ids = set(parent_lookup.keys())
+    
+    for _id, entry in data.items():
+        candidates = set(entry["parents"]) & parent_ids
+        if len(candidates) > 1:
+            print(f"Multiple candidates as overlay map for {_id}: {candidates}. "
+                  f"Currently this is not supported! Skipping ...")
+        elif (len(candidates) == 1) ^ (_id in parent_ids):  # xor
```

Comment:
So what I mean with this line is to check when for one entry there could be multiple available overlay map groups.
E.g. if there were overlay maps for `zentralgelaende` and additonally specific overlay maps for `0501` (building part of the Zentralgelände). Then both would be candidates as overlay maps for a room in `0501`.

There are certainly ways to handle this case (e.g. choose the lowest level parent with overlay maps, e.g. `0501` in this case), but they're not implemented here, because I'm not even sure whether it makes sense to create such high level (`zentralgelaende`) overlay maps.

Including both wouldn't make sense in the UI imo.

Reply:
But yes, below the `if` above it could only be 0 or 1.